outlook/calendar/pkg/graph/calendar.go

```diff
@@ -82,22 +82,20 @@ func ListCalendars(ctx context.Context, client *msgraphsdkgo.GraphServiceClient)
 
 	// TODO - handle if there are more than 100
 
-	if err != nil {
-		return nil, fmt.Errorf("failed to get group memberships: %w", err)
-	}
-
-	for _, group := range memberOf.GetValue() {
-		result, err := client.Groups().ByGroupId(util.Deref(group.GetId())).Calendar().Get(ctx, nil)
-		if err != nil {
-			// Some groups don't have calendars and will just error out. That's fine.
-			continue
+	if err == nil { // if user's account has access to groups feature
+		for _, group := range memberOf.GetValue() {
+			result, err := client.Groups().ByGroupId(util.Deref(group.GetId())).Calendar().Get(ctx, nil)
+			if err != nil {
+				// Some groups don't have calendars and will just error out. That's fine.
```

Comment:
Hmm what is the actual fix? look like the code will do the same behavior?

Reply:
except the logic here:
```
if err != nil {
		return nil, fmt.Errorf("failed to get group memberships: %w", err)
	}
```
Personal accounts do not have access to group features, so this should not be treated as an error.